engine/engine/content/builtins/scripts/mobdebug.lua

```diff
@@ -555,6 +557,56 @@ local function normalize_path(file)
   return (file:gsub("^(/?)%.%./", "%1"))
 end
 
+local function get_filename(file)
+  -- technically, users can supply names that may not use '@',
+  -- for example when they call loadstring('...', 'filename.lua').
+  -- Unfortunately, there is no reliable/quick way to figure out
+  -- what is the filename and what is the source code.
+  -- If the name doesn't start with `@`, assume it's a file name if it's all on one line.
+  if find(file, "^[@=]") or not find(file, "[\r\n]") then
+    file = gsub(gsub(file, "^[@=]", ""), "\\", "/")
+    -- normalize paths that may include up-dir or same-dir references
+    -- if the path starts from the up-dir or reference,
+    -- prepend `basedir` to generate absolute path to keep breakpoints working.
+    -- ignore qualified relative path (`D:../`) and UNC paths (`\\?\`)
+    if find(file, "^%.%./") then file = basedir..file end
+    if find(file, "/%.%.?/") then file = normalize_path(file) end
+    -- need this conversion to be applied to relative and absolute
+    -- file names as you may write "require 'Foo'" to
+    -- load "foo.lua" (on a case insensitive file system) and breakpoints
+    -- set on foo.lua will not work if not converted to the same case.
+    if iscasepreserving then file = string.lower(file) end
+    if find(file, "^%./") then file = sub(file, 3)
+    else file = gsub(file, "^"..q(basedir), "") end
+    -- some file systems allow newlines in file names; remove these.
+    file = gsub(file, "\n", ' ')
+  else
+    file = mobdebug.line(file)
+  end
+  return file
+end
+
+local function get_server_data(file)
```

Comment:
The only custom function I added to be able to receive commands form the server (the editor). This is a small piece of the `debug_hook` function - the default mobdebug's function